tests/unit/query_plan_create_set_remove_delete.cpp

```diff
@@ -1086,3 +1086,499 @@ TEST(QueryPlan, DeleteRemoveProperty) {
   auto context = MakeContext(storage, symbol_table, &dba);
   EXPECT_THROW(PullAll(*rem_op, &context), QueryRuntimeException);
 }
+
+TEST(QueryPlan, SetPropertyWithAuthChecker) {
+  memgraph::storage::Storage db;
+  auto storage_dba = db.Access();
+  memgraph::query::DbAccessor dba(&storage_dba);
+
+  std::string label_name = "l1";
+  auto label = dba.NameToLabel(label_name);
+
+  std::string property_name = "prop";
+  auto property = dba.NameToProperty(property_name);
+  auto property_value = memgraph::storage::PropertyValue{1};
+
+  // Add a single vertex
+  auto v = dba.InsertVertex();
+  ASSERT_TRUE(v.AddLabel(label).HasValue());
+  ASSERT_TRUE(v.SetProperty(property, property_value).HasValue());
+  dba.AdvanceCommand();
+
+  EXPECT_EQ(1, CountIterable(dba.Vertices(memgraph::storage::View::OLD)));
+
+  auto reset_property_value = [&]() { static_cast<void>(v.SetProperty(property, property_value)); };
+
+  auto execute_set_property = [&](memgraph::auth::User user, int new_property_value) {
+    reset_property_value();
+
+    AstStorage storage;
+    SymbolTable symbol_table;
+
+    // MATCH (n) SET n.prop = 2
+    auto scan_all = MakeScanAll(storage, symbol_table, "n");
+
+    auto literal = LITERAL(new_property_value);
+    auto n_p = PROPERTY_LOOKUP(IDENT("n")->MapTo(scan_all.sym_), property);
+    auto set_property = std::make_shared<plan::SetProperty>(scan_all.op_, property, n_p, literal);
+
+    // produce the node
+    auto output =
+        NEXPR("n", IDENT("n")->MapTo(scan_all.sym_))->MapTo(symbol_table.CreateSymbol("named_expression_1", true));
+    auto produce = MakeProduce(set_property, output);
+
+    memgraph::glue::FineGrainedAuthChecker auth_checker{user};
+    auto context = MakeContextWithFineGrainedChecker(storage, symbol_table, &dba, &auth_checker);
+
+    PullAll(*produce, &context);
+    dba.AdvanceCommand();
+  };
+
+  auto execute_set_properties = [&](memgraph::auth::User user, int new_property_value) {
+    reset_property_value();
+
+    AstStorage storage;
+    SymbolTable symbol_table;
+
+    // MATCH (n) SET n = {prop: 2};
+    auto scan_all = MakeScanAll(storage, symbol_table, "n");
+
+    auto prop = PROPERTY_PAIR(property_name);
+    std::unordered_map<PropertyIx, Expression *> prop_map;
+    prop_map.emplace(storage.GetPropertyIx(prop.first), LITERAL(new_property_value));
+    auto *rhs = storage.Create<MapLiteral>(prop_map);
+    auto set_properties =
+        std::make_shared<plan::SetProperties>(scan_all.op_, scan_all.sym_, rhs, plan::SetProperties::Op::UPDATE);
+
+    auto output =
+        NEXPR("n", IDENT("n")->MapTo(scan_all.sym_))->MapTo(symbol_table.CreateSymbol("named_expression_1", true));
+    auto produce = MakeProduce(set_properties, output);
+
+    memgraph::glue::FineGrainedAuthChecker auth_checker{user};
+    auto context = MakeContextWithFineGrainedChecker(storage, symbol_table, &dba, &auth_checker);
+
+    PullAll(*produce, &context);
+    dba.AdvanceCommand();
+  };
```

Comment:
Can you extract these as functions since they are used in other test cases as well